### PR TITLE
Fix useInteractOutside to work with pointer events again

### PR DESCRIPTION
**Pull request by snowystinger**
Closes https://github.com/adobe/react-spectrum/issues/1338
by reverting https://github.com/adobe/react-spectrum/pull/1240

We've merged in https://github.com/adobe/react-spectrum/pull/1469 to solve the original bug that reverting brought back

The new stories in here show that Clicking on an element outside the overlay that makes use of usePress still causes onInteractOutside to fire.
`double menu fires onInteractOutside` can have both menus open, that's being solved in a different PR. This should just be there to show that onInteractOutside fires when clicking on another trigger.
`trigger visible through underlay` action button shouldn't get any events if the underlay is up and you try to click it through the underlay
Testing instructions for the above found here: https://github.com/adobe/react-spectrum/issues/919

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-spectrum/button/stories/ActionButton.stories.tsx [hunk 1]
```diff
@@ -96,3 +100,31 @@ function render(props = {}) {
     </Flex>
   );
 }
+
+// https://codesandbox.io/s/eloquent-herschel-pgwci?file=/index.html
+function DisabledButtonBug() {
+  let [fooDisabled, setFooDisabled] = useState<boolean>();
+  let fooPressStart = () => {
+    console.log('press start foo');
```
Comment (LFDanLu) [inline packages/@react-spectrum/button/stories/ActionButton.stories.tsx:108]: nit: what do you think about replacing the console.logs here w/ `action('fooPressStart')()`? I find it hard to remember what stories fire console logs vs actions when testing

**Review by LFDanLu — COMMENTED** [review]
LGTM, tested on Chrome/FF/Safari, just one small nit and a question

File: packages/@react-spectrum/button/stories/ActionButton.stories.tsx (hunk 1 above)
Comment (snowystinger) [reply packages/@react-spectrum/button/stories/ActionButton.stories.tsx:108]: that's fair, i can do it to the rest of them

File: packages/@react-aria/interactions/src/useInteractOutside.ts [hunk 2]
```diff
@@ -45,12 +45,13 @@ export function useInteractOutside(props: InteractOutsideProps) {
...
-    /*
-    // FF bug https://bugzilla.mozilla.org/show_bug.cgi?id=1675846 prevents us from using this pointerevent
-    // once it's fixed we can uncomment
+
     // Use pointer events if available. Otherwise, fall back to mouse and touch events.
     if (typeof PointerEvent !== 'undefined') {
       let onPointerUp = (e) => {
+        if (isDisabled) {
+          return;
+        }
```
[the remark this answers is not in the document]
Comment (snowystinger) [reply packages/@react-aria/interactions/src/useInteractOutside.ts:54]: it was added to the uncommented section, but never to pointer events
https://github.com/adobe/react-spectrum/commit/88af318fc7898a1e1ffafc40d10cce50e10544b0#diff-36b1df47ccb01e20a099200e27889c7708c475395b5e8cb1a8a3efbc69f54228

**Review by LFDanLu — APPROVED** [review]
LGTM

**Comment by devongovett** [comment]
I can still reproduce in FF 87 as long as the input inside the dialog isn't focused when you click outside.

**Review by LFDanLu — APPROVED** [review]
LGTM, tested in FF, Safari, Chrome desktop + android

**Review by devongovett — APPROVED** [review]

**Review by LFDanLu — APPROVED** [review]

**Pull request merged**